okhttp/src/jvmMain/kotlin/okhttp3/Cookie.kt

```diff
@@ -326,25 +359,38 @@ class Cookie private constructor(
       this.httpOnly = true
     }
 
+    fun sameSite(sameSite: String) = apply {
+      val validSameSites = listOf("lax", "strict", "none")
+      require(validSameSites.contains(sameSite.lowercase())) { "SameSite must be one of: ${validSameSites.joinToString(", ")}" }
+
+      this.sameSite = sameSite
+    }
+
     fun build(): Cookie {
+      if (sameSite.lowercase() == "none") {
```

Comment:
```suggestion
      if (sameSite.equalsIgnoreCase("None")) {
```

This avoids an allocation

Reply:
Updated using Kotlin syntax: `String.equals(value, ignoreCase = true)`